src/cluster.c

```diff
@@ -1048,6 +1058,15 @@ getNodeByQuery(client *c, struct serverCommand *cmd, robj **argv, int argc, int
         numkeys = getKeysFromCommand(mcmd, margv, margc, &result);
         keyindex = result.keys;
 
+        if (mcmd->proc == selectCommand) {
+            long long id;
+            if (getLongLongFromObject(margv[i], &id) != C_OK || (id < 0 || id >= server.dbnum)) {
```

Comment:
This isn't complete verification the command would succeed, this command could fail for other reasons like ACLs.